From 07ec1782299a518e92aff8b3c3359822a8ede5ef Mon Sep 17 00:00:00 2001
From: Calvin Johnson <calvin.johnson@nxp.com>
Date: Wed, 11 Oct 2017 19:23:38 +0530
Subject: [PATCH 320/741] staging: fsl_ppfe/eth: fix RGMII tx delay issue

commit 29a2ec1b97f8e98e0064ee8af019a6a722081408 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Recently logic to enable RGMII tx delay was changed by
below patch.

https://patchwork.kernel.org/patch/9447581/

Based on the patch, appropriate change is made in PFE driver.

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>
Signed-off-by: Anjaneyulu Jagarlmudi <anji.jagarlmudi@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl_ppfe/pfe_eth.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_eth.c b/drivers/staging/fsl_ppfe/pfe_eth.c
index d9aea916c10f..d26ab5fbb0f8 100644
--- a/drivers/staging/fsl_ppfe/pfe_eth.c
+++ b/drivers/staging/fsl_ppfe/pfe_eth.c
@@ -1068,7 +1068,8 @@ static void pfe_eth_adjust_link(struct net_device *ndev)
 			new_state = 1;
 			gemac_set_speed(priv->EMAC_baseaddr,
 					pfe_get_phydev_speed(phydev));
-			if (priv->einfo->mii_config == PHY_INTERFACE_MODE_RGMII)
+			if (priv->einfo->mii_config ==
+					PHY_INTERFACE_MODE_RGMII_TXID)
 				pfe_set_rgmii_speed(phydev);
 			priv->oldspeed = phydev->speed;
 		}
-- 
2.17.1

